src/main/java/com/scaler/lld/design/assignments/builder/DatabaseConfigurationBuilder.java

```diff
@@ -1,6 +1,95 @@
 package com.scaler.lld.design.assignments.builder;
 
+import lombok.Getter;
+import lombok.Setter;
+
 @WithBuilder
+
 public class DatabaseConfigurationBuilder {
 
+    private String databaseUrl;
+    private String username;
+    private String password;
+    private int maxConnections;
+    private boolean enableCache;
+    private boolean isReadOnly;
+
+    public String getDatabaseUrl() {
+        return databaseUrl;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public int getMaxConnections() {
+        return maxConnections;
+    }
+
+    public boolean isEnableCache() {
+        return enableCache;
+    }
+
+    public boolean isReadOnly() {
+        return isReadOnly;
+    }
```

Comment:
[MINOR] You can use the `@Getter` annotation here as well.

Reply:
added